[microTVM] Add wrapper for creating project using a MLF

gromero opened the pull request:
Currently there is already a wrapper function for creating a new project
directory based on an ExportableModule, but there isn't one for creating
a new project directory based on an existing MLF archive, which is also
handy. Hence that commit adds a new wrapper for creating a project using
an existing model compiled and kept in a MLF archive.

Signed-off-by: Gustavo Romero <gustavo.romero@linaro.org>

gromero commented:
cc @areusch

Mousius commented on python/tvm/micro/project.py:
```diff
@@ -154,3 +154,36 @@ def generate_project(
+def generate_project_from_mlf(
+    template_project_dir: typing.Union[pathlib.Path, str],
+    project_dir: typing.Union[pathlib.Path, str],
+    mlf: typing.Union[pathlib.Path, str],
```
```suggestion
    mlf_path: typing.Union[pathlib.Path, str],
```

Mousius commented on python/tvm/micro/project.py:
```diff
@@ -154,3 +154,36 @@ def generate_project(
+    project_dir : pathlib.Path or str
+        Path to a directory where the project will be created.
+
+    mlf : pathlib.Path or str
```
```suggestion
    mlf_path : pathlib.Path or str
```

Mousius commented on python/tvm/micro/project.py:
```diff
@@ -154,3 +154,36 @@ def generate_project(
+    mlf: typing.Union[pathlib.Path, str],
+    options: dict,
+):
+    """Generate a project from a platform template and an existing MLF.
```
```suggestion
    """Generate a project from a platform template and an existing Model Library Format (MLF) archive.
```

Mousius reviewed (changes requested):
This looks pretty straight forward @gromero - a few tweaks and then it LGTM

gromero commented:
@Mousius Thank for the review! All changes done in https://github.com/apache/tvm/pull/9090/commits/7bdd26ca5b2e2dcd69136a8107f0ccf2245a1cb3 

Cheers

areusch commented:
@gromero thanks, please fix the CI issue

gromero commented:
@areusch @Mousius Thanks for the reviews. Sorry for the respin delay. I've fixed the offending line to satisfy the linter and also fixed the type hints for the added function (s/typing.Union/Union/) I think CI haven't caught it because the code is not being used currently -- I found that when using the code on my tvmc micro work. Hopefully https://github.com/apache/tvm/pull/9090/commits/7f6da2c167c053ad11c32464c48515595d0a4bec will be all right for the CI. Cheers.

leandron commented:
Thanks @gromero @Mousius and @areusch - this is merged now.

gromero commented:
@leandron thanks!
